Mathlib/Order/SuccPred/Archimedean.lean

```diff
@@ -68,6 +68,19 @@ theorem Succ.rec_iff {p : α → Prop} (hsucc : ∀ a, p a ↔ p (succ a)) {a b
   obtain ⟨n, rfl⟩ := h.exists_succ_iterate
   exact Iterate.rec (fun b => p a ↔ p b) (fun c hc => hc.trans (hsucc _)) Iff.rfl n
 
+lemma le_total_of_ge {r v₁ v₂ : α} (h₁ : r ≤ v₁) (h₂ : r ≤ v₂) : v₁ ≤ v₂ ∨ v₂ ≤ v₁ := by
```

Comment:
I don't like this naming convention. I think this `r ≤ v₁`, `r ≤ v₂` pattern should be called something like "codirected" or something. Maybe we should even have a predicate so that we can write
```suggestion
lemma le_total_of_codirectedAt {v₁ v₂ : α} (h : CodirectedAt v₁ v₂) : v₁ ≤ v₂ ∨ v₂ ≤ v₁ := by
```

Reply:
Is this predicate useful? In what contexts would you use it?

Reply:
I would think it shows up in eg polytope theory, where it corresponds to "belongs to a common face".

Reply:
Should I add it in this PR, or should I make a separate one?

Reply:
Probably a separate one since it has nothing to do with succ orders. Sorry for atomising all your PRs 💥🙈

Reply:
Looking at this again, I don't see the point of a predicate here, and having `h : ∃ r, r ≤ v₁ ∧ r ≤ v₂` also doesn't seem useful, I think I'll just change the names